src/computer.cpp

```diff
@@ -1260,6 +1261,169 @@ SHORTLY. TO ENSURE YOUR SAFETY PLEASE FOLLOW THE BELOW STEPS. \n\
             query_any( _( "Press any key..." ) );
             break;
 
+        //irradiates food at t_floor_blue, adds radiation
+        case COMPACT_IRRADIATOR: {
+            g->u.moves -= 30;
+            bool error = false;
+            for( const tripoint &dest : g->m.points_in_radius( g->u.pos(), 10 ) ) {
+                if( g->m.ter( dest ) == t_floor_blue ) {
+                    if( g->m.i_at( dest ).empty() ) {
+                        print_error( _( "ERROR: Processing platform empty." ) );
+                    } else {
+                        g->u.moves -= 300;
+                        for( auto it = g->m.i_at( dest ).begin(); it != g->m.i_at( dest ).end(); ++it ) {
+                            // actual food processing
+                            if( !it->rotten() && item_controller->has_template( "irradiated_" + it->typeId() ) ) {
+                                it->convert( "irradiated_" + it->typeId() );
+                            }
+                            // critical failure #1 - let's cook a mininuke
+                            if( it->typeId() == "mininuke" || it->typeId() == "mininuke_act" ) {
+                                print_error( _( "WARNING [409]: Radiation spike detected!" ) );
+                                print_error( _( "  >> AUTO-RISK-ASSESMENT: Geiger profiling..." ) );
+                                print_error( _( "  >> [Self-sustaining] reaction in [critical mass] detected!\n" ) );
+                                print_error( _( "WARNING [999]: Catastrofic meltdown imminent!\n" ) );
+                                print_error( _( "EMERGENCY PROCEDURE [1]: EVACUATE! EVACUATE! EAVACUATE!" ) );
+                                add_msg( m_warning, _( "Evacuate Immediately!" ) );
+                                sounds::sound( g->u.pos(), 30, sounds::sound_t::alarm, _( "an alarm sound!" ) );
+                                it->convert( "mininuke_act" );
```

Comment:
Converting `mininuke_act` into `mininuke_act` is safe?

Reply:
Since (as for below conversation) it will be a conventional explosion, this will be purged anyway. I tested the mininuke, and not tested ```mininuke_act``` assuming it's ok, with changed variables/counters.